agent/xds/listeners.go

```diff
@@ -232,6 +232,15 @@ func (s *ResourceGenerator) listenersFromSnapshotConnectProxy(cfgSnap *proxycfg.
 		// We do not match on all endpoints here since it would lead to load balancing across
 		// all instances when any instance address is dialed.
 		for _, e := range endpoints {
+			if chain.Partition == e.Service.PartitionOrDefault() {
```

Comment:
Ignore my last approval. In failover scenarios (where there may be zero endpoints), this would remove the chain's VIP from the config, right? So we should probably hoist the attachment of the chain outside of this loop.

Reply:
Do we still need this logic here if the code above is always attaching the VIPs?

Reply:
Not sure how that got left in, it's gone now